js/toolbar/z-plane-list.js

```diff
@@ -154,29 +176,85 @@ function ZPlaneList(parent) {
         var loader = _self._zPlaneContainer.querySelector('#z-plane-loader');
         if (loader != null) {
             var carousel = _self._zPlaneContainer.querySelector('.z-plane-carousel');
+            var zIndexInput = _self._zPlaneContainer.querySelector('#main-image-z-index');
+            var zIndexSubmit = _self._zPlaneContainer.querySelector('#z-index-jump-button');
             if (display == true) {
                 carousel.style.pointerEvents = 'none';
                 loader.style.display = 'flex';
-
+                zIndexInput.disabled = true;
+                zIndexSubmit.disabled = true;
             } else {
                 carousel.style.pointerEvents = 'initial';
                 loader.style.display = 'none';
+                zIndexInput.disabled = false;
+                zIndexSubmit.disabled = false;
             }
         }
     };
 
+    /**
+     * change the currentZPlaneRequest variable and fetch the data
+     * @param {string} requestType - must be present in validFetchRequests variable
+     * @param {object} requestData - the content of the fetch request. Make sure the requestData is valid, no such check is done in this function
+     */
+    this._updateCurrentZPlaneRequestAndFetchData = function (requestType, requestData) {
+        _self.currentZPlaneRequest = {
+            type: requestType,
+            data: requestData,
+        };
+
+        if (_self.validFetchRequests.includes(requestType)) {
```

Comment:
`includes` support is limited. Let's just do `_self.validFetchRequests.indexOf(requestType) != -1`

Reply:
Done